Use prim::TensorExprGroup interned symbol

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43743 Update aliasing in tensorexpr fuser
* #43742 use types in the IR instead of vmap
* #43636 Add inline call to profiling executor pipeline
* **#43635 Use prim::TensorExprGroup interned symbol**
* #43634 Update requires grad property

Intern the symbol, no functional changes. Aliasing need to be looked at but this should be done in a separate PR; this PR is just changing the symbol.

Differential Revision: [D23358806](https://our.internmc.facebook.com/intern/diff/D23358806)

Commit:
Use prim::TensorExprGroup interned symbol

[ghstack-poisoned]

Commit:
Update on "Use prim::TensorExprGroup interned symbol"




[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/ir/alias_analysis.cpp:
```diff
@@ -501,6 +503,7 @@ void AliasDb::analyzeImpl(Node* node) {
     case prim::DictConstruct:
     case prim::ListConstruct:
       return analyzeContainerConstruct(node);
+    // TODO: think more about TensorExpr alias correctness
```
Nit: Did you want to emphasize this point by repeating it or is it just a duplicated comment? :)

eellison replied:
duplicated comment, good catch

ZolotukhinM reviewed (approved):
👌

Commit:
Update on "Use prim::TensorExprGroup interned symbol"


Intern the symbol, no functional changes. Aliasing need to be looked at but this should be done in a separate PR; this PR is just changing the symbol.

Differential Revision: [D23358806](https://our.internmc.facebook.com/intern/diff/D23358806)

[ghstack-poisoned]

Commit:
Update on "Use prim::TensorExprGroup interned symbol"


Intern the symbol, no functional changes. Aliasing need to be looked at but this should be done in a separate PR; this PR is just changing the symbol.

Differential Revision: [D23358806](https://our.internmc.facebook.com/intern/diff/D23358806)

[ghstack-poisoned]

Commit:
Update on "Use prim::TensorExprGroup interned symbol"


Intern the symbol, no functional changes. Aliasing need to be looked at but this should be done in a separate PR; this PR is just changing the symbol.

Differential Revision: [D23358806](https://our.internmc.facebook.com/intern/diff/D23358806)

[ghstack-poisoned]

Commit:
Update on "Use prim::TensorExprGroup interned symbol"


Intern the symbol, no functional changes. Aliasing need to be looked at but this should be done in a separate PR; this PR is just changing the symbol.

Differential Revision: [D23358806](https://our.internmc.facebook.com/intern/diff/D23358806)

[ghstack-poisoned]

Commit:
Update on "Use prim::TensorExprGroup interned symbol"


Intern the symbol, no functional changes. Aliasing need to be looked at but this should be done in a separate PR; this PR is just changing the symbol.

Differential Revision: [D23358806](https://our.internmc.facebook.com/intern/diff/D23358806)

[ghstack-poisoned]
